tests/system/feature_store/test_spark_engine.py

```diff
@@ -903,3 +934,185 @@ def test_get_offline_features_with_filter(self):
+            engine="spark",
+        )
+        measurements.graph.to(
+            MapValues(
+                mapping={
+                    "bad": {"ranges": {"one": [0, 30], "two": [30, "inf"]}},
+                    "hr_is_error": {"false": "0", "true": "1"},
+                },
+                # with_original_features=with_original,
+            )
+        )
+        source = ParquetSource("myparquet", path=self.get_remote_pq_source_path())
+        targets = [CSVTarget(name="csv", path=csv_path_spark)]
+        fs.ingest(
+            measurements,
+            source,
+            targets,
+            spark_context=self.spark_service,
+            run_config=fs.RunConfig(local=False),
+        )
+        csv_path_spark = measurements.get_target_path(name="csv")
+
+        measurements = fs.FeatureSet(
+            "measurements_storey",
+            entities=[fs.Entity(key)],
+            timestamp_key="timestamp",
+        )
+        measurements.graph.to(
+            MapValues(
+                mapping={
+                    "bad": {"ranges": {"one": [0, 30], "two": [30, "inf"]}},
+                    "hr_is_error": {"false": "0", "true": "1"},
+                },
+                # with_original_features=with_original,
+            )
+        )
+        source = ParquetSource("myparquet", path=self.get_remote_pq_source_path())
+        targets = [CSVTarget(name="csv", path=csv_path_storey)]
+        fs.ingest(
+            measurements,
+            source,
+            targets,
+        )
+        csv_path_storey = measurements.get_target_path(name="csv")
+        read_and_assert(csv_path_spark, csv_path_storey)
+
+    def test_ingest_with_steps_extractor(self):
+        key = "patient_id"
+        csv_path_spark = "v3io:///bigdata/test_ingest_to_csv_spark"
+        csv_path_storey = "v3io:///bigdata/test_ingest_to_csv_storey.csv"
+
+        measurements = fs.FeatureSet(
+            "measurements_spark",
+            entities=[fs.Entity(key)],
+            timestamp_key="timestamp",
+            engine="spark",
+        )
+        measurements.graph.to(
+            DateExtractor(
+                parts=["day_of_year"],
+                timestamp_col="timestamp",
+            )
+        )
+        source = ParquetSource("myparquet", path=self.get_remote_pq_source_path())
+        targets = [CSVTarget(name="csv", path=csv_path_spark)]
+        fs.ingest(
+            measurements,
+            source,
+            targets,
+            spark_context=self.spark_service,
+            run_config=fs.RunConfig(local=False),
+        )
+        csv_path_spark = measurements.get_target_path(name="csv")
+
+        measurements = fs.FeatureSet(
+            "measurements_storey",
+            entities=[fs.Entity(key)],
+            timestamp_key="timestamp",
+        )
+        measurements.graph.to(
+            DateExtractor(
+                parts=["day_of_year"],
+                timestamp_col="timestamp",
```

Comment:
Test once without providing the `timestamp_col` to make sure the default is handled properly.